sympy/core/expr.py

```diff
@@ -813,11 +813,11 @@ def _eval_interval(self, x, a, b):
             return B - A
 
         value = B - A
-        if all([a.is_comparable, b.is_comparable]):
+        if a.is_comparable and b.is_comparable:
             singularities = list(solveset(self.cancel().as_numer_denom()[1], x))
             for s in singularities:
-                if a < s < b:
-                    value += -limit(self, x, s, "+") + limit(self, x, s, "-")
+                    if a < s < b or b < s < a:
```

Comment:
> How does interchange of a and b cause the singularity to be evaluated differently ?

It seems to me that if `A - B` is returned instead of `B - A`, then also the contributions from singularities should change sign. (If `integrate(1/x**2, (x, -1, 1))` is `oo`, then `integrate(1/x**2, (x, 1, -1)` should be `-oo`.)  At least, I think that this is the common convention about integrals with inverted limits.